api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EntityAgeAccelerator.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.dhsc.htbhf.claimant.testsupport;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entity.BaseEntity;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Capable of accelerating the ageing of entities in the database, by modifying all timestamp properties to put them further in the past.
+ */
+@Component
+public class EntityAgeAccelerator {
+
+    private static final String BASE_PACKAGE_NAME = "uk.gov.dhsc.htbhf";
+    private static final Map<Class, List<Field>> LOCAL_DATE_FIELDS = new HashMap<>();
+    private static final Map<Class, List<Field>> LOCAL_DATE_LISTS = new HashMap<>();
+    private static final Map<Class, List<Field>> LOCAL_DATETIME_FIELDS = new HashMap<>();
+    private static final Map<Class, List<Field>> LOCAL_DATETIME_LISTS = new HashMap<>();
+    private static final Map<Class, List<Field>> NON_ENTITY_CHILD_OBJECT_FIELDS = new HashMap<>();
+    private static final Map<Class, List<Field>> NON_ENTITY_CHILD_OBJECT_LISTS = new HashMap<>();
+
+    /**
+     * Adjusts all LocalDate and LocalDateTime fields on the given entity, and on all child objects of that entity.
+     * A 'child object' is considered to be the value of any field whose type is in a package starting 'uk.gov.dhsc.htbhf',
+     * excluding objects extending {@link BaseEntity} (to prevent infinite recursion & entities being fast-forwarded twice).
+     *
+     * @param entity       the object to fast-forward.
+     * @param numberOfDays the number of days to fast-forward by.
+     */
+    public static void fastForward(Object entity, int numberOfDays) {
+        if (entity == null) {
+            return;
+        }
+        adjustLocalDates(entity, numberOfDays);
```

Comment:
Rather than handling `LocalDate` and `LocalDateTime` separately, we could just look for fields which are of type `Temporal`.